uxarray/_scrip.py

```diff
@@ -1,11 +1,121 @@
-from warnings import warn
+import xarray as xr
+import numpy as np
 
 
-def _read_scrip(filepath):
-    """This is just a function placeholder.
+def _to_ugrid(in_ds, outfile):
+    """If input dataset (ds) file is an unstructured SCRIP file, function will
+    reassign SCRIP variables to UGRID conventions in output file (outfile).
 
-    Not implemented.
+    Parameters
+    ----------
+    in_ds : :class:`xarray.Dataset`
+        Original scrip dataset of interest being used
+
+    outfile : :class:`xarray.Variable`
+        file to be returned by _populate_scrip_data, used as an empty placeholder file
+        to store reassigned SCRIP variables in UGRID conventions
+    """
+
+    if in_ds['grid_area'].all():
+
+        # Create Mesh2_node_x/y variables from grid_corner_lat/lon
+        # Turn latitude scrip array into 1D instead of 2D
+        corner_lat = in_ds['grid_corner_lat'].values
+        corner_lat = corner_lat.flatten()
+
+        # Return only the index of unique values to preserve order
+        lat_idx = np.unique(corner_lat, return_index=True)[1]
+        sort_lat = sorted(lat_idx)
+
+        # Create placeholder array for the sorted unique values
+        unique_lat = np.zeros(len(sort_lat))
+        for i in range(len(sort_lat) - 1):
+            unique_lat[i] = (corner_lat[sort_lat[i]])
+
+        # Repeat above steps with longitude data instead
+        corner_lon = in_ds['grid_corner_lon'].values
+        corner_lon = corner_lon.flatten()
+        lon_idx = np.unique(corner_lon, return_index=True)[1]
+        sort_lon = sorted(lon_idx)
+
+        unique_lon = np.zeros(len(sort_lon))
+        for i in range(len(sort_lon) - 1):
+            unique_lon[i] = (corner_lon[sort_lon[i]])
+
+        # Create Mesh2_node_x/y from unsorted, unique grid_corner_lat/lon
+        outfile['Mesh2_node_x'] = unique_lon
+        outfile['Mesh2_node_y'] = unique_lat
+
+        # Create Mesh2_face_x/y from grid_center_lat/lon
+        outfile['Mesh2_face_x'] = in_ds['grid_center_lon']
+        outfile['Mesh2_face_y'] = in_ds['grid_center_lat']
+
+        # Create array using matching grid corners to create face nodes
+        face_arr = []
+        for i in range(len(in_ds['grid_corner_lat'] - 1)):
+            x = in_ds['grid_corner_lon'][i].values
+            y = in_ds['grid_corner_lat'][i].values
+            face = np.hstack([x[:, np.newaxis], y[:, np.newaxis]])
+            face_arr.append(face)
+
+        face_node = np.asarray(face_arr)
+
+        outfile['Mesh2_face_nodes'] = xr.DataArray(
```

Comment:
The dimensionality of this variable does not seem correct. It should be:

`integer Mesh2_face_nodes(nMesh2_face, nMaxMesh2_face_nodes)`

Please see details [here](https://ugrid-conventions.github.io/ugrid-conventions/#:~:text=the%20current%20convention.-,Example%3A,-dimensions%3A%0AnMesh2_node%20%3D%205).